### PR TITLE
addons: rhel kernel version matching

### PR DESCRIPTION
**Pull request by snir911**
According to https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/precompiled/ the correct way to install pre-compiled drivers is to update kernel and opt-into a precompiled modularity stream, however, it may take some time between latest kernel version is released until the precompiled pkgs are available; which may cause driver failure.
This commit adds a function which tries to update the kernel to the latest version supported by the nvidia precompiled drivers, if fails it upgrades to the latest kernel version available.

### CODE REVIEW

File: src/cloud-api-adaptor/podvm/addons/nvidia_gpu/setup.sh [hunk 1]
```diff
@@ -34,11 +55,11 @@ if  [[ "$PODVM_DISTRO" == "ubuntu" ]]; then
 fi
 if  [[ "$PODVM_DISTRO" == "rhel" ]]; then
     dnf config-manager --add-repo http://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo
```
Comment (beraldoleal) [inline src/cloud-api-adaptor/podvm/addons/nvidia_gpu/setup.sh:57]: The repository is added twice. Here and at line 30, right?

**Review by beraldoleal — COMMENTED** [review]
Hi @snir911, thanks for this. I let one small comment for you.

File: src/cloud-api-adaptor/podvm/addons/nvidia_gpu/setup.sh (hunk 1 above)
Comment (beraldoleal) [reply src/cloud-api-adaptor/podvm/addons/nvidia_gpu/setup.sh:57]: "It doesn't hurt", much. But I would still vote to remove it.

If for some reason, we decide to update one and forget about the other we might have conflicts. Also this is fetching the repository file from the internet twice.

I'm missing the reasoning for keep duplicated code. Do you mind to expand on that?

Comment (snir911) [reply src/cloud-api-adaptor/podvm/addons/nvidia_gpu/setup.sh:57]: FWIW my reasoning was that leaving only one repo configuration will result in sensitivity to the calls order, I'll change it as you're also not the first one mentioning it :)

**Review by littlejawa — APPROVED** [review]
lgtm
Thanks @snir911 !

**Review by beraldoleal — APPROVED** [review]
Except the CI errors, LGTM, thanks.

**Comment by snir911** [comment]
I find it safe to merge this one as this commit affects RHEL podvm image builds which uses the GPU addon only, hence, it has  limited impact on any other image builds.

**Comment by stevenhorsman** [comment]
Snir has informed me that this PR is urgent for downstream, and given the limited scope of it as mentioned as we have a bug to resolve in peer pods before release already (#1941), I think it's okay to merge now.

**Pull request merged**